dali/python/nvidia/dali/_utils/callbacks.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from nvidia.dali import types
+from nvidia.dali import tensors
+np = None
+
+def import_numpy():
+    """Import numpy lazily, need to define global `np = None` variable"""
+    global np
+    if np is None:
+        try:
+            import numpy as np
+        except ImportError:
+            raise RuntimeError('Could not import numpy. Please make sure you have numpy '
+                               'installed before you use parallel mode.')
+
+
+class _SourceKind(Enum):
```

Comment:
I'm not an expert here, but I think that's a bit of a misuse of private names. It's not private to the module if you're importing it. If you want to keep the name private in the importing module, I'd say it's better to use `from _utils.callbacks import SourceKind as _SourceKind`.

Reply:
It's not private to the module, but it's private to DALI, but you're probably right with the correct usage, will adjust.

Reply:
done